src/Maestro/maestro-angular/src/app/page/channel-graph/channel-graph.component.ts

```diff
@@ -0,0 +1,136 @@
+import { Component, Input, OnChanges, SimpleChanges, AfterContentInit } from '@angular/core';
+import { graphlib, render } from 'dagre-d3';
+import { select } from 'd3';
+
+import { FlowGraph, FlowRef, FlowEdge } from 'src/maestro-client/models';
+import { RepoNamePipe } from 'src/app/pipes/repo-name.pipe';
+import { analyzeAndValidateNgModules } from '@angular/compiler';
+
+function getRepositoryShortName(repo:string): string {
+  return `${RepoNamePipe.prototype.transform(repo)}`;
+}
+function getNodeLabel(node:FlowRef): string {
+  let label = `${getRepositoryShortName(node.repository)}\n`;
+  label = `${label}${node.branch}\n`;
+  return label;
+}
+
+function getNodeTitle(node:FlowRef): string {
+  let official = node.officialBuildTime == 0 ? "No successful runs in the last 7 days" : node.officialBuildTime.toFixed(2);
+  let pr = node.prBuildTime == 0 ? "No successful runs in the last 7 days" : node.prBuildTime.toFixed(2);
+
+  let title = `Repository: ${getRepositoryShortName(node.repository)}\n`;
+  title = `${title}Branch: ${node.branch}\n`;
+  title = `${title}Official Build: ${official}\n`;
+  title = `${title}Dep Flow: ${pr}\n`;
+  title = `${title}Best Case Path Time: ${node.bestCaseTime.toFixed(2)}\n`;
+  title = `${title}Worst Case Path Time: ${node.worstCaseTime.toFixed(2)}`;
```

Comment:
Can this just be one giant string that just has literal newlines in it?

Reply:
Yeah, it can be. It was just sort of gross looking. I'll change it.